workshops/platformer/README.md

```diff
@@ -0,0 +1,717 @@
+
+```js
+player = createSprite(100, height-75, 50, 50);
+```
+
+### Making our Player Move
+
+Let's have the player move continuously by adding this line to `draw()`, before `drawSprites();`:
+
+```js
+player.position.x = player.position.x + 5;
+```
+
+Don't forget to redraw the background so the illusion is complete. Place the following line at the top of the `draw()` function:
+
+```js
+background(150, 200, 250);
+```
+
+Save and refresh live preview. Whoops! Your player disappears right off the screen!
+
+### Using Camera to Track Player Movement
+
+We can use the camera to follow the player's movement. p5.play offers the functionality of a camera, which is automatically created. We just reference it with `camera`.
+
+Let's try setting the position of the camera to the position of the player. This will make the camera always follow the player. Add this line directly beneath the line that modifies player movement.
+
+```js
+camera.position.x = player.position.x;
+```
+
+Save and refresh to see the player in action!
+
+### Adjusting Camera ???
+
+Having the player at the center of the screen limits our visual of the obstacles.
+
+We can adjust the camera x-position by setting it slightly ahead of the player, so that we get more screen real estate. Modify the previous line by adding to the camera x-position:
+
+```js
+camera.position.x = player.position.x + (width/4);
+```
+
+### Adjusting the Ground
+
+What we'd like to do is modify the position of the first ground sprite so that it immediately follows the last ground sprite
+
+![](img/DIAGRAM OF INDIAN SPRINTS TODO)
+
+We can select the first ground sprite and store it in a variable by typing the following line in the `draw()` function. I put this logic beneath the line that sets camera x-position.
+
+```js
+var firstGroundSprite = groundSprites[0];
+```
+
+`[0]` refers to the thing in the Group at index 0.
+
+![](img/ARRAYS AND INDEXING TODO)
+
+Next, we'll check if it has moved off screen, by comparing its x-position with the left edge of the camera.
+
+The x-position of the camera is `camera.position.x`. The x-position of the left edge of the camera is `camera.position.x - width/2`. Remember, the camera is at the middle of the screen.
+
+![](img/PLACEMENT RELATIVE TO CAMERA TODO)
+
+We'll set up a conditional to check if the x-position of the ground sprite is less than that of the left edge of the camera.
+
+We will put this conditional directly beneath the definition of the firstGroundSprite variable;
+
+```js
+var firstGroundSprite = groundSprites[0];
+if (firstGroundSprite.position.x <= camera.position.x - (width/2)) {
+
+}
+```
+
+If it is indeed off-screen, then we must remove it from the Group, alter its x-position, and append it to the Group at the last index.
```

Comment:
A little confusing.  How can you alter it's position if it no longer exists? Maybe something like 
"...remove it from the Group, alter the x position of the now saved GroundSprite, then append firstGroundSprite at the last index" idk i'm not a word smith but I feel this could be better worded.